src/json_protocol.rs

```diff
@@ -258,11 +258,41 @@ pub fn build_file_upload_request(hash: &[u8], media_type: MediaType) -> JsonValu
     }, base64::encode(hash)]
 }
 
+pub fn build_media_conn_request() -> JsonValue {
+    array!["query", "mediaConn"]
+}
+
 pub fn parse_file_upload_response<'a>(response: &'a JsonValue) -> Result<&'a str> {
     parse_response_status(response)?;
     response.get_str("url")
 }
 
+pub fn parse_media_conn_response<'a>(response: &'a JsonValue) -> Result<(&'a str, i64, Vec<&'a str>)> {
+    parse_response_status(response)?;
+
+    match response["media_conn"] {
```

Comment:
If the `media_conn` field isn't present in the `response`, this seems like it would panic. Maybe try `response.get()` or something?

Reply:
If response is not a object or if the response does not contain a field "media_conn", then a NULL Json Value is returned. No panic here.
I took some freedom on the returned error, though.